backend/tests/test_rate_limit_fix.py

```
@@ -0,0 +1,926 @@
+            rate_limit.window_start = now
+            rate_limit.window_end = now + window_duration
+        
+        # Verify limit_value is unchanged after reset
+        assert rate_limit.limit_value == configured_limit, \
+            "limit_value should remain unchanged after window reset"
+        assert rate_limit.current_usage == 0, \
+            "current_usage should be reset to 0"
+
+    def test_consistency_between_check_and_increment(self):
+        """Test that check_rate_limit and increment_rate_limit use same limit value"""
+        tenant_id = uuid.uuid4()
```

Ruff (0.14.7)

Local variable `tenant_id` is assigned to but never used

Remove assignment to unused variable `tenant_id`

(F841)